pilot/pkg/bootstrap/server.go

```diff
@@ -343,9 +343,15 @@ func NewServer(args *PilotArgs, initFuncs ...func(*Server)) (*Server, error) {
 	// Notice that the order of authenticators matters, since at runtime
 	// authenticators are activated sequentially and the first successful attempt
 	// is used as the authentication result.
-	authenticators := []security.Authenticator{
-		&authenticate.ClientCertAuthenticator{},
+	authenticators := []security.Authenticator{}
+	// Add XFCC authenticator first in the list if it needs to be used for authentication.
+	// Otherwise, the ClientCertAuthenticator will reject the plain text calls to Istiod when
```

Comment:
How does it reject things? We have a list on authenticatores, and if any return non-err we return the result, otherwise we return all errors.

The inverse of this comment is also true, if XFCC is first then it will reject things without XFCC?

Reply:
> The inverse of this comment is also true, if XFCC is first then it will reject things without XFCC?

Correct. But XFCC is only enabled under certain conditions which is 
- Trusted Gateway CIDR is set
- PLAIN TEXT auth is enabled
in that case, the intent is really to just to XFCC  - So IMO it is OK to reject there.

But what Costin brings up also makes sense like if any of the authenticators is true. We can allow.

Reply:
Same confusion as @howardjohn 

We should set the most commonly used authenticator at first place

Reply:
> But what Costin brings up also makes sense like if any of the authenticators is true. We can allow.

This is the default behavior